solidity/test/maintainer/Maintainer.test.ts

```diff
@@ -0,0 +1,3577 @@
+      let tx: ContractTransaction
+      let initThirdPartyBalance: BigNumber
+
+      before(async () => {
+        await createSnapshot()
+
+        initThirdPartyBalance = await provider.getBalance(thirdParty.address)
+
+        await maintainerProxy.connect(governance).authorize(thirdParty.address)
+        await reimbursementPool
+          .connect(governance)
+          .authorize(maintainerProxy.address)
+
+        tx = await maintainerProxy
+          .connect(thirdParty)
+          .requestNewWallet(activeWalletMainUtxo)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should emit NewWalletRequested event", async () => {
+        await expect(tx).to.emit(bridge, "NewWalletRequested")
+      })
+
+      it("should refund ETH", async () => {
+        const postNotifyThirdPartyBalance = await provider.getBalance(
+          thirdParty.address
+        )
+        const diff = postNotifyThirdPartyBalance.sub(initThirdPartyBalance)
+
+        expect(diff).to.be.gt(0)
+        expect(diff).to.be.lt(
+          ethers.utils.parseUnits("2000000", "gwei") // 0,002 ETH
+        )
+      })
+    })
+  })
+
+  describe("submitDepositSweepProof", () => {
+    context("when called by an unauthorized third party", async () => {
+      const data: DepositSweepTestData = SingleP2SHDeposit
+      // Take wallet public key hash from first deposit. All
+      // deposits in same sweep batch should have the same value
+      // of that field.
+      const { walletPubKeyHash } = data.deposits[0].reveal
+
+      before(async () => {
+        await createSnapshot()
+
+        await bridge.setWallet(walletPubKeyHash, {
+          ecdsaWalletID: ethers.constants.HashZero,
+          mainUtxoHash: ethers.constants.HashZero,
+          pendingRedemptionsValue: 0,
+          createdAt: await lastBlockTime(),
+          movingFundsRequestedAt: 0,
+          closingStartedAt: 0,
+          pendingMovedFundsSweepRequestsCount: 0,
+          state: walletState.Live,
+          movingFundsTargetWalletsCommitmentHash: ethers.constants.HashZero,
+        })
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        // Set the deposit dust threshold to 0.0001 BTC, i.e. 100x smaller than
+        // the initial value in the Bridge in order to save test Bitcoins.
+        await bridge.setDepositDustThreshold(10000)
+
+        const sweepOutcome: Promise<ScenarioOutcome> = runSweepScenario(data)
+
+        await expect(sweepOutcome).to.be.revertedWith(
+          "Caller is not authorized"
+        )
+      })
+    })
+
+    context("when called by an authorized party", async () => {
+      before(async () => {
+        await createSnapshot()
+
+        await maintainerProxy.connect(governance).authorize(thirdParty.address)
+        await reimbursementPool
+          .connect(governance)
+          .authorize(maintainerProxy.address)
```

Comment:
Instead of repeating this logic in setup functions for different tests, can we instantiate `authorizedMaintainer` unnamed signer in the main setup and authorize it there?

For all cases of `when called by an unauthorized third party` we would use a `thirdParty` and for all cases of `when called by an authorized maintainer`, we would use an `authorizedMaintainer`.

Reply:
[e4cbc78](https://github.com/keep-network/tbtc-v2/pull/209/commits/e4cbc7812021ef1e8b0cfa97f616d6991d587948)